crates/subspace-service/src/domains.rs

```diff
@@ -156,24 +171,55 @@ where
 
                 match response {
                     Ok(response) => {
-                        trace!("Response from a peer {peer_id},",);
+                        trace!(%attempt, "Response from a peer {peer_id},",);
 
-                        return Some(response.last_confirmed_block_receipt);
+                        let receipt = response.last_confirmed_block_receipt;
+                        let receipt_hash = KeccakHasher::hash(&receipt.encode());
+
+                        peers_hashes.insert(*peer_id, receipt_hash);
+                        receipts.insert(receipt_hash, receipt);
+                        receipts_hashes
+                            .entry(receipt_hash)
+                            .and_modify(|count: &mut u32| *count += 1)
+                            .or_insert(1u32);
                     }
                     Err(error) => {
-                        debug!("Domain info request failed. peer = {peer_id}: {error}");
+                        debug!(%attempt, "Domain info request failed. peer = {peer_id}: {error}");
 
                         continue 'peers;
                     }
                 }
             }
             debug!(
                 domain_id=%self.domain_id,
-                "No synced peers to handle the domain confirmed block infor request. Pausing..."
+                "No synced peers to handle the domain confirmed block info request. Pausing..."
             );
 
+            if peers_hashes.len() >= PEERS_THRESHOLD {
+                break;
+            }
+
             sleep(REQUEST_PAUSE).await;
         }
+
+        if peers_hashes.len() < PEERS_THRESHOLD {
+            debug!(peers=%peers_hashes.len(), "Couldn't pass peer threshold for receipts.");
+        }
+
+        // Find the receipt with the maximum votes
+        if let Some(max_receipt_vote) = receipts_hashes.values().max() {
+            if let Some((receipt_hash, _)) = receipts_hashes
+                .iter()
+                .find(|(_, vote)| max_receipt_vote == *vote)
```

Comment:
It's different from a similar segment header protocol because we can't return several last confirmed domain blocks, can we? What if I make several request rounds to the same peer at some interval and possibly get several confirmed blocks?

Reply:
>What if I request the receipt twice from the connected peers with an interval of 5 seconds and compare the aggregated results?

That may be helpful but is still probabilistic.

I have an idea that also needs 2 requests from the peer:
1.  Request the receipt from the peer as it is
2. Send the smallest confirmed receipt to the peer, the peer after receiving the receipt only return `true` if
    - The peer after received the receipt it only return `true` if a) the receipt is indeed confirmed and b) its local view of the last confirmed receipt is 
4.